src/aof.c

```diff
@@ -160,15 +186,135 @@ void aof_background_fsync(int fd) {
     bioCreateBackgroundJob(REDIS_BIO_AOF_FSYNC,(void*)(long)fd,NULL,NULL);
 }
 
+#ifdef USE_NVML
+/* Open additional PMEMlog file, which will be used for rewrites, both files
+ * will be available at launch, when rewrite will be triggered, then rewr pool
+ * will be populated with fresh data, buffer will be appended and pointers of
+ * pools will be exchanged */
+int openAppendOnlyFileREWR(void) {
+    size_t rewr_len;
+    struct redis_stat sb;
+    /* server.aof_filename + '_rewr' + '\0' */
+    rewr_len = strlen(server.aof_filename) + 6;
+    server.aof_filename_rewr = (char*) zmalloc(rewr_len * sizeof(char));
+    snprintf(server.aof_filename_rewr, rewr_len, "%s_rewr", server.aof_filename);
+
+    if (redis_stat(server.aof_filename_rewr,&sb) != -1) {
+        if ((server.aof_plp_rewr = pmemlog_open(server.aof_filename_rewr)) == NULL) {
+            redisLog(REDIS_WARNING, "Redis cannot open NVML AOF REWR file: %s", strerror(errno));
+            exit(1);
+        }
+    } else {
+        if ((server.aof_plp_rewr = pmemlog_create(server.aof_filename_rewr, server.aof_nvml_log_size, 0600)) == NULL) {
+            redisLog(REDIS_WARNING, "Redis cannot create NVML AOF REWR file: %s", strerror(errno));
+            return REDIS_ERR;
+        }
+    }
+
+    pmemlog_rewind(server.aof_plp_rewr);
```

Comment:
No need to rewind if this is a new pool - move it to the if() part (after line 206).